modulemd/foreman_modulemd.yaml

```diff
@@ -0,0 +1,18 @@
+---
+document: modulemd
+version: 2
+data:
+    name: foreman
+    stream: latest
```

Comment:
Thinking about versioning: does it make sense to introduce the version number?

Reply:
That depends on the user workflow we want. As I understand it, keeping the stream the same means you can have more seamless like experience. I would honestly need to play around a bit more with this by creating two versions with two versions of Foreman and see.

Reply:
Mostly thinking out loud here.

I suppose that if you use `latest`, the upgrade workflow would be essentially the same as we have today: you change the yum repository and then run `dnf upgrade` and all the additional steps.

With a versioned approach you can keep the multiple repositories but it also becomes possible to have a single repository. With both you would still need an explicit instruction to tell dnf to use the new module.

Having a versioned approach also means the Katello module can properly depend on the correct Foreman version.

I also wonder what happens if the module would switch from Ruby 2.7 to 3.0. Can that happen within `latest` or are separate versions better?

A single repository has the benefit of no longer needing to manage the `yum.repos.d` file. However, it would also mean our metadata grows forever. Our current pipelines are also not at all ready for a single repository.

Another thought: can we have a `latest` and `latest-staging` module? I'm not sure if it makes sense.

So in short:
1. current yum.theforeman.org layout
  1.1. `latest` only
  1.2. explicitly state the version
2. Introduce a single modular repository with multiple module versions

Option 2 is unrealistic in the short term and it's questionable if it's desirable at all in the long term.

Reply:
Explicitly stating the version means the user will have to perform wild `module reset` & `module enable` dances when upgrading between versions. At no benefit (as our repo will never offer multiple streams from the same repo).

Reply:
as for switching `ruby:2.7` to `ruby:3.0`, you can change that in the `requires`, but the user will need to perform a manual `dnf module reset` before this takes action.

Reply:
before that (`dnf module reset`) happens, no packages from that module are available to install, so the installer can't "fix" it for us :/

Reply:
This is very helpful in understanding the implications for users. I appreciate it.